src/kernel/include/shared/cos_types.h

```diff
@@ -266,6 +284,19 @@ enum
 	BOOT_CAPTBL_FREE = round_up_to_pow2(BOOT_CAPTBL_LAST_CAP, CAPMAX_ENTRY_SZ)
 };
 
+/*
+ * The slots of the sinv to user-level fault handlers.
+ */
+typedef enum {
+	FAULT_CAPTBL_MEM_ACCESS        = BOOT_CAPTBL_FLT_MEM_ACCESS,
```

Comment:
I feel like `COMP_CAPTBL_FAULT_XXXX` would probably be a better name for these fault capabilities.

I think your point about booter having those fault capabilities is somewhat tricky and perhaps will only make sense to have a subset of fault caps.
If there is a page fault in booter and booter is invoked to resolve it, that literally would be a double fault. So, I'm not sure if it's essential for us to do that. 
But in some projects like fswmanager that Yuxin, Vlad were working on, booter is not just a booter there. (I don't know if that's a long-term idea). In those cases, we'd probably have some scope for booter to have some of the fault capabilities.  In general, if booter is overloaded, then having those fault SINVs in booter would be useful and not otherwise IMO.